### PR TITLE
feat: Integrate tile server

### PR DESCRIPTION
**Pull request by drduhe**
**Issue #, if available:** n/a

### Notes
- We are introducing a new feature, Tile Server. What is it? The OversightML Tile Server is a lightweight, cloud-based tile server which allows you to quickly pass an image from S3 bucket to get metadata, image statistics, and set of tiles in real-time.
  - Github: https://github.com/aws-solutions-library-samples/osml-tile-server
  - PyPi: https://pypi.org/project/osml-tile-server
  - Github Registry: https://github.com/aws-solutions-library-samples/osml-tile-server/pkgs/container/osml-tile-server
  - DockerHub: https://hub.docker.com/repository/docker/awsosml/osml-tile-server/general

### Testing
- Fully deployed to my personal dev account and ran integration tests against it
- Validate if all the checks succeeded in Github actions 

Before you submit a pull request, please make sure you have to following:

- [x] Your code contains tests that cover all new code and changes
- [x] All new and existing tests passed
- [x] I have read the [Contributing Guidelines](https://github.com/aws-solutions-library-samples/guidance-for-overhead-imagery-inference-on-aws/blob/main/CONTRIBUTING.md) and agree to follow the [Code of Conduct](
https://github.com/aws-solutions-library-samples/guidance-for-overhead-imagery-inference-on-aws/blob/main/CODE_OF_CONDUCT.md))

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by RanbirAulakh** [comment]
Github Workflow will fail until the `osml-tile-server` is public. Otherwise, we won't be able to perform pull. Once the `osml-tile-server` goes public, we should update line 9 in `.github/codebuild/build.yml` from

> if [[ "${BASE_REF}" -eq "dev" ]]; then jq '(.buildModelContainer, .buildAppContainer) = true' lib/accounts/target_account_template.json > lib/accounts/target_account.json; fi

to 

> if [[ "${BASE_REF}" -eq "dev" ]]; then jq '(.buildFromSource) = true' lib/accounts/target_account_template.json > lib/accounts/target_account.json; fi

Assuming we have `buildFromSource` in `OSMLAccount.ts` in `osml-cdk-constructs`

File: lib/accounts/target_account_template.json [hunk 1]
```diff
@@ -3,9 +3,6 @@
   "name": "OSML",
   "region": "us-west-2",
   "prodLike": false,
-  "buildAppContainer": false,
-  "buildModelContainer": false,
-  "enableAutoscaling": true,
-  "enableMonitoring": true,
-  "enableTesting": true
+  "deployModelRunner": true,
+  "deployTileServer": true
```
[the remark this answers is not in the document]
Comment (drduhe) [reply lib/accounts/target_account_template.json:7]: Let's require this be configured at the stack level.

File: lib/osml-stacks/model_runner/mr-imagery.ts [hunk 2]
```diff
@@ -4,13 +4,13 @@
...
 import { IVpc } from "aws-cdk-lib/aws-ec2";
-import { MRImagery, MRSMRole, OSMLAccount } from "osml-cdk-constructs";
+import { MEContainer, MRImagery, OSMLAccount } from "osml-cdk-constructs";
 
 export interface MRImageryStackProps extends StackProps {
   env: Environment;
   account: OSMLAccount;
   vpc: IVpc;
-  mrSmRole?: MRSMRole;
+  mrSmRole?: MEContainer;
```
[the remark this answers is not in the document]
Comment (drduhe) [reply lib/osml-stacks/model_runner/mr-imagery.ts:13]: Removed

File: lib/osml-stacks/model_runner/mr-roles.ts [hunk 3]
```diff
@@ -15,31 +15,31 @@ export interface MRRolesStackProps extends StackProps {
  */
 export class MRRolesStack extends Stack {
   public mrTaskRole: MRTaskRole;
-  public mrSmRole: MRSMRole;
+  public mrSmRole: MESMRole;
```
[the remark this answers is not in the document]
Comment (drduhe) [reply lib/osml-stacks/model_runner/mr-roles.ts:18]: Updated.

**Review by RanbirAulakh — APPROVED** [review]
LGTM!

File: README.md [hunk 4]
```diff
@@ -77,9 +78,8 @@ This package uses a number of tools to enforce formatting, linting, and general
        "name": <unique name for stacks>,
        "region": <target region for deployment>,
        "prodLike": <false || true marks resource retention>
-       "enableAutoscaling": <false || true enable autoscaling>,
-       "enableMonitoring": <false || true enable monitoring dashboards>,
-       "enableTesting": <false || true enable testing infrastructure>
+       "deployModelRunner": <false || true deploy model runner>,
```
[the remark this answers is not in the document]
Comment (drduhe) [reply README.md:82]: Fixed

**Comment by RanbirAulakh** [comment]
MRRole must be created first then VpcStack because when it coms to destroying it, it will delete vpc first then roles. Since VPC is created first, then roles. Ultimately, in the end, it destroys the roles then vpc which causes this issue.

**Pull request merged**